temporal-sdk/src/main/java/io/temporal/internal/common/ProtoEnumNameUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.common;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.collect.ImmutableMap;
+import io.temporal.api.enums.v1.IndexedValueType;
+import io.temporal.api.enums.v1.WorkflowTaskFailedCause;
+import java.util.Map;
+
+/**
+ * Because in Go enums are in a shared namespace, protobuf best practices is to prefix the enums
+ * with enum type prefix to create unique names. The final result is verbose and when we use a
```

Comment:
Well, in this AIP:
"The other values should not be prefixed by the name of the enum itself. This generally requires users to write MyState.MYSTATE_ACTIVE in their code, which is unnecessarily verbose."
But this is also right under:
"To avoid sharing values, APIs may prefix enum values with the name of the enum. In this case, they must do so consistently within the enum."